src/export_edf.jl

```diff
@@ -97,7 +97,7 @@ function onda_samples_to_edf_header(samples::AbstractVector{<:Samples};
 end
 
 function onda_samples_to_edf_signals(onda_samples::AbstractVector{<:Samples}, seconds_per_record::Float64)
-    edf_signals = Union{EDF.AnnotationsSignal,EDF.Signal}[]
+    edf_signals = Union{EDF.AnnotationsSignal,EDF.Signal{Int16}}[]
```

Comment:
What's the benefit to adding this parameter restriction? I could be missing something, but seems like this is an implementation detail and I don't see a correctness/performance benefit to surface/codify it?

for example, it feels like this code shouldn't break if the `EDF.Signal` constructor decided to e.g. promote provided encoded sample data to Int32 (not that that would happen, just kinda making a hypothetical point)

Reply:
This `Union` is the type of the `signals` field in an `EDF.File` object now that `EDF.Signal` is parameterized on the encoded sample type (for BDF support in EDF.jl v0.7). I think the purpose of adding it here is that you won't be able to construct an `EDF.File` programmatically with a `Union{EDF.AnnotationsSignal,EDF.Signal}`, since `convert` that's called automatically with inner constructors will fail.